components/eventing-controller/controllers/subscription/beb/reconciler.go

```diff
@@ -643,18 +657,25 @@ func (r *Reconciler) syncInitialStatus(ctx context.Context, subscription *eventi
 	}
 	// case: conditions are already initialized
 	if len(currentStatus.Conditions) >= len(expectedStatus.Conditions) && !updateReadyStatus {
```

Comment:
is that even a valid check? if `currentStatus.Conditions  == [a,b,c]` and `expectedStatus.Conditions == [d,e]` that would pass, but clearly expected conditions would not be in currentStatus

how about always processing the conditions and not return early here?

Reply:
This was the old behavior, I didn't want to change it.
Changed now.